platform/barefoot/docker-syncd-bfn/base_image_files/monit_syncd

```diff
@@ -3,5 +3,5 @@
 ## process list:
 ##  syncd
 ###############################################################################
-check process syncd matching "/usr/bin/syncd\s"
-    if does not exist for 5 times within 5 cycles then alert
+check program syncd|syncd with path "/usr/bin/process_checker syncd syncd /usr/bin/syncd --diag"
```

Comment:
Why did you go back to adding the `--diag` flag? The reason we originally changed to `"/usr/bin/syncd\s"` was so that we wouldn't need to worry about which flags were passed to the process.

Reply:
Good catch! Initially I used the string `/usr/bin/syncd\s` as a parameter of the script `process_checker`. In the line 38 of the script `process_checker`, I checked whether the string `/usr/bin/syncd\s` is `in` the process command line. If I used `/usr/bin/syncd\s` then it will did not match the syncd process, then I went back to use the string `/usr/bin/syncd --diag`.

After you asked me this question, I thought again and found I can use the regular expression in python to do this check. For example, I can still use the string `/usr/bin/syncd\s` as a parameter of the script. In the line 38, `re.search(process_cmdline, ' '.join(process.cmdline()))` can be used to do this check.

Reply:
I'm not sure we even need to use regex here. The original problem was that with `check process`, Monit would search for the string anywhere in the process command line. Now that we're using /usr/bin/process_checker, we have control over the logic. You can adjust the logic to simply ensure that the command starts with the string, then we don't need the regex.

Reply:
OK, Let me give an example to double check what I am understanding is correct. For example, for `syncd` process, the script `/usr/bin/process_checker` will accept the string `/usr/bin/sync\s` as a parameter. In the line 38, we can use the `' '.join(process.cmdline()).startswith(process_cmd.replace("\s", " "))` method to check whether that process is `syncd` process or not, right?

Reply:
No. I'm saying that `/usr/bin/process_checker` can take `/usr/bin/syncd` as a parameter, and it will only try to find a process whose command line _starts with_ that string. Unlike Monit, which would try to find any process whose command line _contains_ that string.

Reply:
Got it. That's the difference between Monit and `/usr/bin/process_checker` to find a process. For example, for `syncd` process, `/usr/bin/process_checker` will accept '/usr/bin/syncd' as the parameter and then this script can use the method `(' '.join(proceess.cmdline)).startswith(process_cmdline)` to search any process whose command line start with '/usr/bin/syncd', right?

Reply:
Right.

Reply:
Joe, Thanks for your great suggestions such that I can polish this script.